utils/time_slots_utils.js

```diff
@@ -0,0 +1,84 @@
+const {models} = require('../db/relations')
+const TimeSlot = models.TimeSlot
+
+const TimeSlotsUtils = {
+
+    getIntersection: function (slots) {
```

Comment:
Tu skorzystam z uwagi Radka, że nie każda linijka kodu w inżynierce musi mieć podpis, że coś tu robiłem XD. Ogólnie nie rozumiem, co tu się dzieje, ale może to dlatego, że dalej ciula potrafię z ASD, więc zostawiam nieprzeczytane (ale ufam, że działa) XDDD

Reply:
Ogólnie faktycznie wygląda w ciul jak jakieś zadanie z ASD więc przywołuje naszego algorytmicznego koksa @Pinioo

Reply:
rozwiązanie z neta lekko modyfikowane, sam nie do końca czuje czemu to działa XD

Reply:
To filtrowanie mi trochę śmierdzi, ale tak to wygląda git. No i trza dodać, że dana tablica slots[i] musi być posortowana (ale no nie ma powodu, żeby nie była posortowana)

Reply:
sortuje w serwisie przed wołaniem tej funkcji